rpc/client/eventstream/eventstream.go

```diff
@@ -0,0 +1,188 @@
+// Package eventstream implements a convenience client for the Events method
+// of the Tendermint RPC service, allowing clients to observe a resumable
+// stream of events matching a query.
+package eventstream
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/tendermint/tendermint/rpc/coretypes"
+)
+
+// Client is the subset of the RPC client interface consumed by Stream.
+type Client interface {
+	Events(ctx context.Context, req *coretypes.RequestEvents) (*coretypes.ResultEvents, error)
+}
+
+// ErrStopRunning is returned by a Run callback to signal that no more events
+// are wanted and that Run should return.
+var ErrStopRunning = errors.New("stop accepting events")
+
+// A Stream cpatures the state of a streaming event subscription.
+type Stream struct {
+	filter     *coretypes.EventFilter // the query being streamed
+	batchSize  int                    // request batch size
+	newestSeen string                 // from the latest item matching our query
+	waitTime   time.Duration          // the long-polling interval
+	client     Client
+}
+
+// New constructs a new stream for the given query and options.
+// If opts == nil, the stream uses default values as described by
+// StreamOptions.  This function will panic if cli == nil.
+func New(cli Client, query string, opts *StreamOptions) *Stream {
+	if cli == nil {
+		panic("eventstream: nil client")
```

Comment:
:(

Reply:
This is completely not worth reporting an error for.